src/main/java/prefs/Preferences.java

```diff
@@ -128,15 +138,32 @@ public void setLastOpenBoard(String board) {
     }
     
     public Optional<String> switchBoard() {
-        if (getLastOpenBoard().isPresent() && getAllBoards().size() > 1) {
-            List<String> boardNames = getAllBoardNames();
+        setBoardToSwitch();
+        return getLastOpenBoard();
+    }
+
+    /**
+     * Sets the lastOpenBoard to the board to switch to
+     */
+    private void setBoardToSwitch(){
```

Comment:
Here's a rough design I would use.

1. `UI` would contain a `Boards` class, which can be serialised. It keeps track of board state (which is open, unsaved changes, etc.) and has high-level operations like creating a new board, etc. It isn't concerned with how boards are persisted.
2. We represent boards directly with a `Board` class, which has a name, `List<String>` of filters, etc. and is also meant to be serialised.
3. `Boards` knows about `PanelControl` (to create panels), and `MenuControl` knows about `Boards` (as a UI to board operations). We can look into decoupling them in future, but right now it doesn't seem worth the trouble of adding a lot of new events/callbacks/interfaces.
4. `Preferences` knows about `Boards` and decides when and how to serialise it. `Boards` controls what to serialise. Not clear on the details here as preferences are being changed too.

@garbanzos feel free to improve on this!

There are quite a few related PRs now, which might lead to conflicts. #1255 (board equality) and #1224 (unsaved changes) depend on this now. #1272 will change how preferences work, and #1262 changes the related tests. We can continue to discuss the redesign, but maybe we should defer implementation until a few more of the rest are done?